sql/parser/expr.go

```diff
@@ -241,72 +241,156 @@ func (NullVal) String() string {
 
 // QualifiedName is a base name and an optional indirection expression.
 type QualifiedName struct {
-	Base     Name
-	Indirect Indirection
+	Base       Name
+	Indirect   Indirection
+	tableName  bool
```

Comment:
Splitting into separate types is problematic due to the grammar changes. (I actually started out trying that approach and quickly gave up). Changing to an enum. It started as a single bool and then evolved.